lib/livebook_web/live/hub/new_live.ex

```diff
@@ -151,8 +184,123 @@ defmodule LivebookWeb.Hub.NewLive do
     """
   end
 
+  defp disabled_class(true), do: "opacity-30 pointer-events-none"
+  defp disabled_class(false), do: ""
+
+  defp card_item_border_class(id, id), do: "border-gray-200"
+  defp card_item_border_class(_, _), do: "border-gray-100"
+
+  defp card_item_class(id, id), do: "bg-gray-200"
+  defp card_item_class(_, _), do: "bg-gray-100"
+
+  @impl true
+  def handle_event("select_option", %{"value" => option}, socket) do
+    {:noreply,
+     socket
+     |> assign(selected_option: option)
+     |> assign_form(option)}
+  end
+
+  def handle_event("validate", %{"new_org" => attrs}, socket) do
+    changeset =
+      socket.assigns.org
+      |> Teams.change_org(attrs)
+      |> Map.replace!(:action, :validate)
+
+    {:noreply, assign_form(socket, changeset)}
+  end
+
+  def handle_event("save", %{"new_org" => attrs}, socket) do
+    case Teams.create_org(socket.assigns.org, attrs) do
+      {:ok, response} ->
+        attrs = Map.merge(attrs, response)
+        changeset = Teams.change_org(socket.assigns.org, attrs)
+        org = Ecto.Changeset.apply_action!(changeset, :insert)
+
+        Process.send_after(self(), {:check_completion_data, org}, 1000)
+
+        {:noreply,
+         socket
+         |> assign(
+           requested_code: true,
+           org: org,
+           verification_uri: response["verification_uri"]
+         )
+         |> assign_form(changeset)}
+
+      {:error, changeset} ->
+        {:noreply, assign_form(socket, changeset)}
+
+      {:transport_error, message} ->
+        {:noreply, put_flash(socket, :error, message)}
+    end
+  end
+
   @impl true
-  def handle_event("select_type", %{"value" => service}, socket) do
-    {:noreply, assign(socket, selected_type: service)}
+  def handle_info({:check_completion_data, %Org{} = org}, socket) do
+    case Teams.get_org_request_completion_data(org) do
+      {:ok, :awaiting_confirmation} ->
+        Process.send_after(self(), {:check_completion_data, org}, 1000)
+
+        {:noreply, socket}
+
+      {:ok, response} ->
```

Comment:
Let's pattern match here:

```suggestion
      {:ok, %{"id" => id, ...}} ->
```

It will be a tiny bit more robust.